scvi/hub/hub_model.py

```diff
@@ -0,0 +1,244 @@
+        adata: Optional[AnnData] = None,
+        use_gpu: Optional[Union[str, int, bool]] = None,
+    ):
+        """Placeholder docstring. TODO complete."""
+        logger.info("Loading model...")
+        # get the class name for this model (e.g. TOTALVI)
+        torch_model = torch.load(self._model_path)
+        cls_name = torch_model["attr_dict"]["registry_"]["model_name"]
+        python_module = importlib.import_module(self.metadata.model_parent_module)
+        model_cls = getattr(python_module, cls_name)
+        if adata is not None or os.path.isfile(self._adata_path):
+            self._model = model_cls.load(
+                os.path.dirname(self._model_path), adata=adata, use_gpu=use_gpu
+            )
+        else:
+            # in this case, we must download the large adata if it exists in the model card; otherwise, we error out
+            # the call below faults in self.adata_large if it is None
+            if self.adata_large is None:
+                raise ValueError(
+                    "Could not find any dataset to load the model with.\
+                    Either provide a dataset on disk or a url to download the data in the model card.\
+                    See scvi-tools tutorials for more details."
+                )
+            else:
+                self._model = model_cls.load(
+                    os.path.dirname(self._model_path),
+                    adata=self.adata_large,
+                    use_gpu=use_gpu,
+                )
+
+    def read_adata(self):
+        """Placeholder docstring. TODO complete."""
+        if os.path.isfile(self._adata_path):
+            logger.info("Reading adata...")
+            self._adata = anndata.read_h5ad(self._adata_path)
+        else:
+            logger.info("No data found on disk. Skipping...")
+
+    def read_adata_large(self):
+        """Download the full adata, if it exists, then read it into memory."""
+        large_data_url = self.metadata.large_data_url
+        if large_data_url is not None:
+            logger.info(
+                f"Downloading large dataset from this url:\n{large_data_url}..."
+            )
+            dn = Path(self._adata_large_path).parent.as_posix()
+            fn = Path(self._adata_large_path).name
+            _download(large_data_url, dn, fn)
+            logger.info("Reading large data...")
+            self._adata_large = anndata.read_h5ad(self._adata_large_path)
+        else:
+            logger.info("No large_data_url found in the model card. Skipping...")
+
+
+def list_all_models(
+    do_print: bool = True, print_detailed: bool = False
+) -> Optional[List[ModelInfo]]:
+    """Placeholder docstring. TODO complete."""
+    filt = ModelFilter(library=HF_LIBRARY_NAME)
+    api = HfApi()
+    all_models = api.list_models(filter=filt)
+    if do_print:
+        for m in all_models:
+            print("\n")
+            logger.info(m.__repr__() if print_detailed else m)
+    return None if do_print else all_models
```

Comment:
this could be a rich table

Reply:
definitely should output a dataframe though

Reply:
can we make a hf space to help query hf?

Reply:
I removed two comments that I thought were dupes of the responses here but now the ones that are here are also gone and I cant seem to be able to undelete those ☹️
Anyway, from memory these were:
- This function should return a df
- Can we have a hf space to query hf

The former is done. For the latter: that's an interesting idea. I've added it to this issue for future consideration: https://github.com/scverse/scvi-tools/issues/1817